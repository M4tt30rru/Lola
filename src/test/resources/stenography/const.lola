##Find
 const ##Identifier(_name) = ##Literal(_value);
 ##run {
if isinstance(_value, str):
 _type = 'String'
elif isinstance(_value, (int, long)):
 _type = 'int'
else:
 _type = 'double'
 }
 ##replace public static final ##(_type) ##(_name) = ##(_value);
##Find
 const ##Identifier(_name) = ##Any(_value);
 ##run {
print _value.name
if _value.name == 'true' or _value.name == 'false':
 _type = 'boolean'
else:
 _type = 'Object'
 }
 ##replace public static final ##(_type) ##(_name) = ##(_value);

const o = new O();
const w = true;
const x = 1;
const y = "1";
const z = 1.0;
