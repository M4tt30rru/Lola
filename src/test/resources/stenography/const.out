public static final Object o = new O();
public static final boolean w = true;
public static final int x = 1;
public static final String y = "1";
public static final double z = 1.0;
